pkgs/os-specific/linux/zfs/generic.nix

```diff
@@ -221,11 +221,9 @@ let
           "--with-linux=${kernel.dev}/lib/modules/${kernel.modDirVersion}/source"
           "--with-linux-obj=${kernel.dev}/lib/modules/${kernel.modDirVersion}/build"
         ]
-        ++ kernelModuleMakeFlags
+        ++ map (f: "KERNEL_${f}") kernelModuleMakeFlags
```

Comment:
This scares me a little bit if we happen to introduce variables that shouldn’t get the `KERNEL_` prefix later, but I guess at worst it’d most likely be harmless.